docs/tutorials/write-your-first-comit-app/04-2-taker-takes-order.md

```diff
@@ -0,0 +1,168 @@
+
+    const takerNegotiator = new TakerNegotiator(
+        taker.comitClient,
+        makerNegotiatorUrl
+    );
+
+    const criteria = {
+        buy: {
+            ledger: "bitcoin",
+            asset: "bitcoin",
+            minNominalAmount: "1",
+        },
+        sell: {
+            ledger: "ethereum",
+            asset: "ether",
+        },
+        minRate: 0.001,
+    };
+
+    const order = await takerNegotiator.getOrder(criteria);
+
+    console.log("Rate offered: ", order.getOfferedRate().toString());
+
+    const swap = await order.take();
+
+    console.log(
+        `Took the following order: %s:%s for a rate of %d:%d`,
+        order.rawOrder.ask.asset,
+        order.rawOrder.bid.asset,
+        order.rawOrder.ask.nominalAmount,
+        order.rawOrder.bid.nominalAmount
+    );
+
+    process.exit();
+})();
+```
+
+Note, that in order to properly retrieve an order at the taker side, the maker application has to run at this stage!
+If your maker app is not running you can start it with `yarn maker`.
+
+Ensure that your maker app is running and then start the taker app with `yarn taker` - it should print:
+```
+yarn run v1.22.0
+$ ts-node ./src/taker.ts
+COMIT Taker app
+[Taker] Bitcoin balance: 10, Ether balance: 1000
+Rate offered:  0.02
+Took the following order: ether:bitcoin for a rate of 50:1
+✨  Done in 2.78s.
+```
+
+Once you run the taker app, the **maker app** will terminate as well:
+```
+yarn run v1.22.0
+$ ts-node ./src/maker.ts
+COMIT Maker app
+[Maker] Bitcoin balance: 10, Ether balance: 1000
+Maker's Negotiation Service is listening on localhost:2318.
+Waiting for someone to take my order at: http://127.0.0.1:2318
+✨  Done in 11.01s.
+```
+
+This is, because the maker was waiting for the incoming swap request on cnd and has received it from the taker.
+When the taker was taking the order he sent a swap request.
+The maker's cnd node registered that and the maker app terminated when it noticed it.
+Since the logic for "waiting on incoming swaps" on the maker side is rather simple (it always choses the first one) the maker app will just terminate when started again (because there is already a swap in the cnd queue).
+
+
+It is important to understand, that the swap actually still exists in the maker cnd and the taker cnd!
+The two cnd nodes were started as part of the dev-environemnt and don't know about the taker and the maker app.
+So, both the maker's cnd node and the taker's cnd node are still waiting for the swap to be executed.
+
+Given our simple program logic, this is a problem now, because we are not detecting that there is a swap already in progress.
+The maker will always pick the first swap, while the taker will (once running through taking an order again) be on a different swap.
+
+> Note by Daniel: How should we solve this? The example is a bit incomplete here... 
+> Should we add matching logic, so it does not pick "the first" swap in the list?
```

Comment:
I'm against this. The logic around this is not that straight forward. You would end up with a fully fledged maker.